File: Core/value/booleanvalue.cpp
#include "booleanvalue.h"

BooleanValue::BooleanValue() : ValueBase() {
}

BooleanValue::BooleanValue(ValueGroup *valueGroup, QString id, VALUE_TYPE valueType, bool alwaysEmit, QObject *parent) : ValueBase (valueGroup, id, valueType, alwaysEmit, QVariant::Bool, parent)
{

}

QVariant BooleanValue::_updateValue(QVariant newValue) {
    if (newValue.canConvert(QVariant::Bool)) {
        return QVariant::fromValue(newValue);
    } else {
        return QVariant();
    }
}

File: AndroidApp/OshCore/src/main/java/com/osh/value/BooleanValue.java
package com.osh.value;

import com.osh.SerializationSupport;

public class BooleanValue extends ValueBase<BooleanValue, Boolean> implements SerializationSupport {

	@Override
	protected Boolean _updateValue(Object newValue) {
		if (newValue instanceof Boolean) {
			return (Boolean) newValue;
		} else if (newValue instanceof Number) {
			return ((Number) newValue).intValue() == 1;
		} else if (newValue instanceof String) {
			return Boolean.getBoolean((String) newValue);
		} else {
			return null;
		}
	}

	public BooleanValue(ValueGroup valueGroup, String id, ValueType valueType) {
		super(valueGroup, id, valueType);
	}

	public BooleanValue() {
		super();
	}

}
